prime-router/src/main/kotlin/cli/TestReportStream.kt

```diff
@@ -91,25 +98,55 @@ Examples:
        """
     )
 
-    private val list by option(
-        "--list",
-        help = "List available tests, then quit."
-    ).flag(default = false)
-
     private val env by option(
         "--env",
-        help = "Specify 'local, 'test', or 'staging'.  'local' will connect to ${ReportStreamEnv.LOCAL.endPoint}," +
+        help = "Specify local, test, staging, or prod.  'local' will connect to ${ReportStreamEnv.LOCAL.endPoint}," +
             " and 'test' will connect to ${ReportStreamEnv.TEST.endPoint}"
-    ).choice("test", "local", "staging").default("local").validate {
+    ).choice("test", "local", "staging", "prod").default("local").validate {
+        envSanityCheck()
         when (it) {
             "test" -> require(!key.isNullOrBlank()) { "Must specify --key <secret> to submit reports to --env test" }
             "staging" ->
                 require(!key.isNullOrBlank()) { "Must specify --key <secret> to submit reports to --env staging" }
+            "prod" -> error("Sorry, prod is not implemented yet")
+        }
+    }
+
+    private val key by option(
+        "--key",
+        metavar = "<secret>",
+        help = "Specify reports function access key"
+    )
+
+    private val dir by option(
+        "--dir",
+        help = "specify a working directory for generated files.  Default is $defaultWorkingDir"
+    ).default(defaultWorkingDir)
+
+    // Avoid accidentally connecting to the wrong database.
+    private fun envSanityCheck() {
+        val dbEnv = System.getenv("POSTGRES_URL") ?: error("Missing database env var. For help:  ./prime --help")
+        val problem: Boolean = when (env) {
+            "staging" -> !dbEnv.contains("pdhstaging")
+            "test" -> !dbEnv.contains("pdhtest")
+            "local" -> !dbEnv.contains("localhost")
+            "prod" -> !dbEnv.contains("pdhprod")
+            else -> true
+        }
+        if (problem) {
+            echo("Error: --env is $env but database is set to $dbEnv")
+            exitProcess(-1)
         }
     }
 
     override fun run() {
-        if (list) doList()
+        DatabaseAccess.isFlywayMigrationOK = false
```

Comment:
Turn off flyway migrations - see my comment above.  Please check my idea carefully - can this go wrong somehow?

Reply:
This appears correct to me for this particular use case.

If we want to cover the situation more generally (we write a new cli class and forget about the need to disable migrations), I think the default should be disabled, with enabled being turned on via an environment variable, and if no environment variable, defaulting to false.

For even more protection, the environment variable could be something like `FLYWAY_MIGRATION_ENVIRONMENT_ENABLED=local` and the environment variable would have to match the name set in `PRIME_ENVIRONMENT`. This way if we accidentally run a config mixing environments, there is an extra layer of protection.

Obviously, the best solution here is to ensure testing is always done with read-only accounts, but I am all for adding multiple layers of protection.

Reply:
Thanks, Ron.  I was reluctant to default false at the moment, because the flyway migrations work so well 99.9% of the time.  We won't enable this in prod until @cglodosky's database hardening is done.